include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -210,6 +210,74 @@ struct BFloat16_t {
 
 static_assert(sizeof(BFloat16_t) == sizeof(uint16_t), "Sizes must match");
 
+#if !defined(DISABLE_FLOAT8_TYPES)
```

Comment:
>DISABLE_FLOAT8_TYPES [](http://example.com/codeflow?start=13&length=20)

This header is generally a customer facing one. Thus they build may or may not include this define. NuGet package may not include it and the customer may still not define it.

Typically, we would not do such a thing in the public headers. Instead, if any of the API is called that involves this type, that API would return an error with a message that the build does not support the type.

However, we avoid such `#ifdefs` in the public API.

Reply:
What about the minimum build? If there is only one public API and it is not available for the minimal build, I can safely remove that. However, I saw that sparse tensors are also inside #if !defined(DISABLE... #endif.

Reply:
I see that too. I will need to remove them, it just does not make sense.